tests/Provider/QueryBuilderProvider.php

```diff
@@ -1225,10 +1227,101 @@ public static function update(): array
                     ':qp0' => '{{test}}',
                 ],
             ],
+            [
+                '{{table}}',
+                ['name' => '{{tmp}}.{{name}}'],
+                [],
+                'tmp',
+                [],
+                static::replaceQuotes(
+                    <<<SQL
+                    UPDATE [[table]] SET [[name]]=:qp0 FROM [[tmp]]
+                    SQL
+                ),
+                [
+                    ':qp0' => '{{tmp}}.{{name}}',
+                ],
+            ],
+            [
+                '{{table}}',
+                ['name' => '{{tmp}}.{{name}}'],
+                [],
+                ['tmp'],
+                [],
+                static::replaceQuotes(
+                    <<<SQL
+                    UPDATE [[table]] SET [[name]]=:qp0 FROM [[tmp]]
+                    SQL
+                ),
+                [
+                    ':qp0' => '{{tmp}}.{{name}}',
+                ],
+            ],
+            [
+                '{{table}}',
+                ['name' => '{{tmp}}.{{name}}'],
+                ['id' => 1],
+                'tmp',
+                [],
+                static::replaceQuotes(
+                    <<<SQL
+                    UPDATE [[table]] SET [[name]]=:qp0 FROM [[tmp]] WHERE [[id]] = 1
+                    SQL
+                ),
+                [
+                    ':qp0' => '{{tmp}}.{{name}}',
+                ],
+            ],
+            [
+                '{{table}}',
+                ['name' => '{{tmp}}.{{name}}'],
+                [],
+                new Expression('SELECT [[name]] FROM [[tmp]] WHERE [[id]] = 1'),
```

Comment:
```suggestion
                static::getDb()->select('name')->from('tmp')->where(['id' => 1]),
```

Let's replace with Query, it is more common case.

Reply:
Done